ci-operator/step-registry/mco/conf/day2/destroy-custom-mcp/mco-conf-day2-destroy-custom-mcp-commands.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+function set_proxy () {
+    if [ -s "${SHARED_DIR}/proxy-conf.sh" ]; then
+        echo "Setting the proxy ${SHARED_DIR}/proxy-conf.sh"
+        # shellcheck source=/dev/null
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "No proxy settings"
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+}
+
+# waiting for mcp become a given status
+function wait_mcp(){
+    local mcp_name="${1}" expected_updated="$2" expected_machine_count="$3" wait_time=30
+    local updated MACHINECOUNT
+    while (( wait_time > 0 )); do
+        sleep 10
+        out="$(oc get machineconfigpools ${mcp_name} --no-headers)"
+        echo $out
+        updated="$(echo "${out}" | awk '{print $3}')"
+        MACHINECOUNT="$(echo "${out}" | awk '{print $6}')"
+
+        if [[ "${updated}" == "${expected_updated}" && "${MACHINECOUNT}" == "${expected_machine_count}" ]]; then
```

Comment:
Compared with `mco-conf-day2-create-custom-mcp`, `mco-conf-day2-create-custom-mcp` is using:
```
oc wait mcp "$mcp_name" --for='condition=UPDATING=True' --timeout=300s
oc wait mcp "$mcp_name" --for='condition=UPDATED=True' --timeout="$mcp_timeout"
```
to check the mcp is being updated, and applied. I think the idea is better than the developed loop logic here, and that can make us have a consistent style for create/destroy custom step.

Of course, in this step, after the above two commands finished, we can run one more check - `"${MACHINECOUNT}" == "${expected_machine_count}"` to make it more robust.

Reply:
done

Reply:
how about step `cucushift-upgrade-eus-unpause-worker-mcp`? unpause mcp can also trigger mcp updating?

Reply:
> unpause mcp can also trigger mcp updating ?

I think so, but the while waiting loop in that step is already existing for a while, I am okay with not modifying it for now, if we need to update it in the future, we can do that.

Reply:
Ignore it, I saw it on https://github.com/openshift/release/pull/55328/files#diff-6da6be9bbf08fa8f36b9c7df36e496571acc8899a4e116eff4494a8a255b9c85R69